rustls/src/crypto/aws_lc_rs/mod.rs

```diff
@@ -224,16 +224,20 @@ static SUPPORTED_SIG_ALGS: WebPkiSupportedAlgorithms = WebPkiSupportedAlgorithms
 /// All defined key exchange groups supported by aws-lc-rs appear in this module.
 ///
 /// [`ALL_KX_GROUPS`] is provided as an array of all of these values.
+/// [`DEFAULT_KX_GROUPS`] is provided as an array of this provider's defaults.
 pub mod kx_group {
     pub use super::kx::{SECP256R1, SECP384R1, X25519};
 }
 
-pub use kx::ALL_KX_GROUPS;
+/// A list of the default key exchange groups supported by this provider.
+pub static DEFAULT_KX_GROUPS: &[&dyn SupportedKxGroup] = kx::ALL_KX_GROUPS;
+
+/// A list of all the key exchange groups supported by this provider.
+pub static ALL_KX_GROUPS: &[&dyn SupportedKxGroup] = kx::ALL_KX_GROUPS;
```

Comment:
What is the reason for redefining `ALL_KX_GROUPS` here rather than redefining as before? Seems like that generates some potential for confusion.

Reply:
This is attempting to avoid a `unreachable_pub` warning down the road. They way that happens is:

- for ring, `kx.rs` is included and has a `pub static ALL_KX_GROUPS ...`, and `pub use`s it directly because it contains all the kx groups. That works.
- for aws-lc-rs, `kx.rs` is included and has a `pub static ALL_KX_GROUPS ...`, but cannot use it directly because there are additional kx groups (the post-quantum ones) that need adding (prepending and appending) to it.

I guess an alternative is to remove `ALL_KX_GROUPS` from `kx.rs`? That doesn't seem so bad.

Reply:
(previously: https://github.com/rustls/rustls/pull/2288#discussion_r1893654949)

Reply:
> I guess an alternative is to remove ALL_KX_GROUPS from kx.rs? 

Did this, and added a note why to the commit message.